moveit_ros/move_group/src/default_capabilities/get_group_urdf_capability.cpp

```diff
@@ -0,0 +1,141 @@
+      GET_URDF_SERVICE_NAME,
+      [this](const std::shared_ptr<moveit_msgs::srv::GetGroupUrdf::Request>& req,
+             const std::shared_ptr<moveit_msgs::srv::GetGroupUrdf::Response>& res) {
+        res->error_code.source = LOG_NAME;
+        const auto subgroup = context_->moveit_cpp_->getRobotModel()->getJointModelGroup(req->group_name);
+        // Check if group exists in loaded robot model
+        if (!subgroup)
+        {
+          const auto error_string = std::string("Cannot create URDF because planning group '") + req->group_name +
+                                    std::string("' does not exist");
+          RCLCPP_ERROR(getLogger(), "%s", error_string.c_str());
+          res->error_code.message = error_string;
+          res->error_code.val = moveit_msgs::msg::MoveItErrorCodes::FAILURE;
+          return;
+        }
+        // Get robot description string
+        std::string full_urdf_string;
+        context_->moveit_cpp_->getNode()->get_parameter_or("robot_description", full_urdf_string, std::string(""));
+
+        // Check if string is empty
+        if (full_urdf_string.empty())
+        {
+          const auto error_string =
+              std::string("Couldn't load the urdf from parameter server. Is the /robot_description parameter "
+                          "initialized?");
+          RCLCPP_ERROR(getLogger(), "%s", error_string.c_str());
+          res->error_code.message = error_string;
+          res->error_code.val = moveit_msgs::msg::MoveItErrorCodes::FAILURE;
+          return;
+        }
+
+        // Create subgroup urdf
+        // Create header
+        res->urdf_string = std::string("<?xml version=\"1.0\" ?>\n<robot name=\"") + req->group_name +
+                           std::string("\" xmlns:xacro=\"http://ros.org/wiki/xacro\">");
+
+        // Create links
+        const auto& link_names = subgroup->getLinkModelNames();
+        for (const auto& link_name : link_names)
+        {
+          const auto start = full_urdf_string.find("<link name=\"" + link_name);
+          auto substring = full_urdf_string.substr(start, full_urdf_string.size() - start);
+          res->urdf_string += substring.substr(0, substring.find(LINK_ELEMENT_CLOSING) + LINK_ELEMENT_CLOSING.size());
+        }
+        // Create joints
+        const auto& joint_names = subgroup->getJointModelNames();
+        for (const auto& joint_name : joint_names)
+        {
+          const auto start = full_urdf_string.find("<joint name=\"" + joint_name);
+          auto substring = full_urdf_string.substr(start, full_urdf_string.size() - start);
+          res->urdf_string += substring.substr(0, substring.find(JOINT_ELEMENT_CLOSING) + JOINT_ELEMENT_CLOSING.size());
+        }
+        // Create closing
+        res->urdf_string += ROBOT_ELEMENT_CLOSING;
+
+        // Validate xml file
+        tinyxml2::XMLDocument group_urdf_xml;
+        group_urdf_xml.Parse(full_urdf_string.c_str());
```

Comment:
Use [parseURDF](https://github.com/ros/urdfdom/blob/29426ec119e9c3a9df9fc705d08c262a97b7f285/urdf_parser/src/check_urdf.cpp#L93) for validation